web/src/containers/PatientDetails/PatientDocumentDetails/index.tsx

```diff
@@ -148,38 +152,60 @@ function PatientDocumentDetailsReadonly(props: {
+                                    remoteData={response}
+                                    renderLoading={() => (
+                                        <Button type="primary" className={s.button} disabled>
+                                            <Trans>History</Trans>
+                                        </Button>
+                                    )}
                                 >
-                                    <Trans>Amend</Trans>
-                                </Button>
+                                    {({ provenanceList }) => (
+                                        <Button
+                                            type="primary"
+                                            onClick={() => navigate(`${location.pathname}/history`)}
+                                            className={s.button}
+                                            disabled={provenanceList.length === 0}
+                                        >
+                                            <Trans>History</Trans>
+                                        </Button>
+                                    )}
+                                </RenderRemoteData>
                             </>
                         ) : null}
                         {canBeEdited ? (
                             <>
+                                <ConfirmActionButton
+                                    action={() => deleteDraft(navigate, patientId, qrId)}
+                                    reload={reload}
+                                    qrId={qrId}
+                                    title={t`Are you sure you want to delete the document?`}
+                                    okText="Yes"
+                                    cancelText="No"
+                                >
+                                    <Button className={s.button} type={'text'} danger>
+                                        <Trans>Delete</Trans>
+                                    </Button>
+                                </ConfirmActionButton>
                                 <Button
-                                    type="link"
+                                    type="primary"
                                     onClick={() => navigate(`${location.pathname}/edit`)}
                                     className={s.button}
                                 >
                                     <Trans>Edit</Trans>
                                 </Button>
-                                <Button
-                                    type="link"
-                                    onClick={() => deleteDraft(navigate, patientId, qrId)}
-                                    className={s.button}
-                                >
-                                    <Trans>Delete</Trans>
-                                </Button>
                             </>
                         ) : null}
                     </div>
```

Comment:
let's add some code for
`web/src/containers/PatientDetails/PatientDocument/usePatientDocument.ts`
```
export interface PatientDocumentData {
    formData: QuestionnaireResponseFormData;
    onSubmit: (formData: QuestionnaireResponseFormData) => Promise<void>;
    provenance?: WithId<Provenance>;
};

export function usePatientDocument(props: Props): {
    response: RemoteData<PatientDocumentData>;
    questionnaireId: string;
} {
    const { questionnaireResponse, questionnaireId, onSuccess } = props;
```

then in
`web/src/containers/PatientDetails/PatientDocumentDetails/index.tsx`
```
function PatientDocumentDetailsFormData(props: {
    questionnaireResponse: WithId<QuestionnaireResponse>;
    patient: WithId<Patient>;
    children: (props: PatientDocumentData) => ReactElement;            // <----- updated line
}) {
    const { questionnaireResponse, children, patient } = props;
    const { response } = usePatientDocument({
        ...props,
        patient: patient,
        questionnaireId: questionnaireResponse.questionnaire!,
    });

    return (
        <RenderRemoteData remoteData={response} renderLoading={Spinner}>
            {(response) => children(response)}                   // <----- updated line
        </RenderRemoteData>
    );
}

....

<PatientDocumentDetailsFormData
                    questionnaireResponse={questionnaireResponse}
                    {...props}
                >
                    {({ formData, provenance }) => (  // <----- updated line
                        <Routes>
                            ......
                                
                                <Route
                                    path="/"
                                    element={
                                        <PatientDocumentDetailsReadonly
                                            formData={formData}
                                            encounter={encounter}
                                            reload={manager.reload}
                                            provenance={provenance}     // <----- updated line
                                        />
                                    }
                                />


```